app/controllers/solutions_controller.rb

```diff
@@ -1,2 +1,14 @@
 class SolutionsController < ApplicationController
-end
+ def create
+    @question = Question.find(params[:question_detail])
+    @solution = Solution.find(params[:solution_detail])
```

Comment:
`@question`と紐づいていません。

```suggestion
    @solution = @question.solutions.build(solution_params)
```

このように書いておいて，ストロングパラメータ`solution_params`を定義しましょう。
書き方は通常の元とほぼ同じですが，`binding.pry`で書き方を確認してみるとよいでしょう。